src/Tickets/Emails/Dispatcher.php

```diff
@@ -0,0 +1,525 @@
+<?php
+/**
+ * Email Dispatcher class.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Tickets\Emails
+ */
+
+namespace TEC\Tickets\Emails;
+
+/**
+ * Class Dispatcher.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Tickets\Emails
+ */
+class Dispatcher {
+
+	/**
+	 * Whether this dispatcher was used.
+	 *
+	 * @since TBD
+	 *
+	 * @var bool
+	 */
+	protected bool $used = false;
+
+	/**
+	 * Stores the Email used in this Dispatcher.
+	 *
+	 * @since TBD
+	 *
+	 * @var Email_Abstract|null
+	 */
+	protected ?Email_Abstract $email = null;
+
+	/**
+	 * Stores the all the headers used on the email.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string,string>
+	 */
+	protected array $headers = [];
+
+	/**
+	 * Stores the all the attachments used on the email.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string>
+	 */
+	protected array $attachments = [];
+
+	/**
+	 * Stores the contents to be dispatched.
+	 *
+	 * @since TBD
+	 *
+	 * @var ?string
+	 */
+	protected ?string $content = null;
+
+	/**
+	 * Stores where we will dispatch to.
+	 *
+	 * @since TBD
+	 *
+	 * @var ?string
+	 */
+	protected ?string $to = null;
+
+	/**
+	 * Stores the subject to be dispatched.
+	 *
+	 * @since TBD
+	 *
+	 * @var ?string
+	 */
+	protected ?string $subject = null;
+
+	/**
+	 * Sets the Email instance that will be used for this Dispatcher, intentionally this is protected method.
+	 * You can only generate a new Dispatcher from the Factory Method.
+	 *
+	 * @since TBD
+	 *
+	 * @param Email_Abstract $email
+	 */
+	protected function set_email( Email_Abstract $email ): void {
+		$this->email = $email;
+	}
+
+	/**
+	 * Gets the email instance for this Dispatcher.
+	 *
+	 * @since TBD
+	 *
+	 * @return Email_Abstract|null
+	 */
+	public function get_email(): ?Email_Abstract {
+		return $this->email;
+	}
+
+	/**
+	 * Prepares the dispatcher for this Email.
+	 *
+	 * @since TBD
+	 *
+	 * @param Dispatcher $dispatcher
+	 *
+	 * @return Dispatcher
+	 */
+	protected function prepare_dispatcher( Dispatcher $dispatcher ): Dispatcher {
```

Comment:
This needs to be `public` so that it is actually callable on line 166.

```suggestion
	public function prepare_dispatcher( Dispatcher $dispatcher ): Dispatcher {
```

Reply:
Not needed, marking as resolved.